apollo-router/src/plugins/rhai/mod.rs

```diff
@@ -1131,40 +1137,40 @@ fn process_error(error: Box<EvalAltResult>) -> ErrorDetails {
                 // Clone is annoying, but we only have a reference, so...
                 let map = obj.clone().cast::<Map>();
 
-                let mut ed_status: Option<StatusCode> = None;
-                let mut ed_message: Option<String> = None;
-
-                let status_opt = map.get("status");
-                let message_opt = map.get("message");
-
-                // Now we have optional Dynamics
-                // Try to process each independently
-                if let Some(status_dyn) = status_opt {
-                    if let Ok(value) = status_dyn.as_int() {
-                        if let Ok(status) = StatusCode::try_from(value as u16) {
-                            ed_status = Some(status);
-                        }
-                    }
-                }
-
-                if let Some(message_dyn) = message_opt {
-                    let cloned = message_dyn.clone();
-                    if let Ok(value) = cloned.into_string() {
-                        ed_message = Some(value);
-                    }
-                }
-
-                if let Some(status) = ed_status {
+                if let Some(status) = map
+                    .get("status")
+                    .and_then(|status_dyn| status_dyn.as_int().ok())
+                    .and_then(|value| StatusCode::try_from(value as u16).ok())
+                {
```

Comment:
I like this. I should always go back and check for useful combinators when I finish writing code...

Reply:
it can take a while to write with the combinators, but then it's a bit nicer to follow what's going on